firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -119,11 +120,14 @@ public Thread newThread(Runnable r) {
       RandomFidGenerator fidGenerator) {
     this.firebaseApp = firebaseApp;
     this.serviceClient = serviceClient;
-    this.executor = executor;
     this.persistedInstallation = persistedInstallation;
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    this.backgroundExecutor = backgroundExecutor;
+    this.networkExecutor =
+        new ThreadPoolExecutor(
+            1, 1, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(), threadFactory);
```

Comment:
Do I read this correctly?
0L Milliseconds is the timeout time?

Questions and Notes:
1) Can we use the same numbers that we use for the backgroundExecutor?
2) Please create constants and use them for both executors
3) Why is the core pool size different to the other executor?

Reply:
0L is the keep alive time - when the number of threads is greater than the core, this is the maximum time that excess idle threads will wait for new tasks before terminating.

Created constants that are shared between the two executors.